tests/utils/cloud/test_persistence.py

```diff
@@ -1,105 +1,110 @@
+            self.storage_client.download_to_file(
+                bucket_name=self.TEST_BUCKET_NAME, path_in_bucket=self.FILENAME, local_path=download_local_path
             )
 
             with open(download_local_path) as f:
                 self.assertTrue('{"height": 32}' in f.read())
 
-    def test_upload_and_download_file_as_string(self):
+    def test_download_as_string(self):
         """Test that a file can be uploaded to Google Cloud storage and downloaded as a string."""
-        storage_client = GoogleCloudStorageClient(project_name=self.PROJECT_NAME)
-        filename = "file_to_upload.txt"
-
         with tempfile.TemporaryDirectory() as temporary_directory:
 
-            upload_local_path = f"{temporary_directory}/{filename}"
+            upload_local_path = f"{temporary_directory}/{self.FILENAME}"
 
             with open(upload_local_path, "w") as f:
                 f.write("This is a test upload.")
 
-            upload_url = storage_client.upload_file(
+            self.storage_client.upload_file(
                 local_path=upload_local_path,
                 bucket_name=self.TEST_BUCKET_NAME,
-                path_in_bucket=filename,
+                path_in_bucket=self.FILENAME,
             )
 
-            self.assertEqual(upload_url, f"https://storage.cloud.google.com/{self.TEST_BUCKET_NAME}/{filename}")
-
         self.assertEqual(
-            storage_client.download_as_string(bucket_name=self.TEST_BUCKET_NAME, path_in_bucket=filename),
+            self.storage_client.download_as_string(bucket_name=self.TEST_BUCKET_NAME, path_in_bucket=self.FILENAME),
             "This is a test upload.",
         )
+
+    def test_delete(self):
+        """Test that a file can be deleted."""
+        self.storage_client.upload_from_string(
+            serialised_data=json.dumps({"height": 32}),
+            bucket_name=self.TEST_BUCKET_NAME,
+            path_in_bucket=self.FILENAME,
+        )
+
+        self.assertEqual(
+            json.loads(
+                self.storage_client.download_as_string(bucket_name=self.TEST_BUCKET_NAME, path_in_bucket=self.FILENAME),
+            ),
+            {"height": 32},
+        )
+
+        self.storage_client.delete(bucket_name=self.TEST_BUCKET_NAME, path_in_bucket=self.FILENAME)
+
+        with self.assertRaises(google.api_core.exceptions.NotFound):
+            self.storage_client.download_as_string(bucket_name=self.TEST_BUCKET_NAME, path_in_bucket=self.FILENAME)
+
+    def test_get_metadata(self):
+        """Test that file metadata can be retrieved."""
+        self.storage_client.upload_from_string(
+            serialised_data=json.dumps({"height": 32}),
+            bucket_name=self.TEST_BUCKET_NAME,
+            path_in_bucket=self.FILENAME,
+        )
+
+        metadata = self.storage_client.get_metadata(
+            bucket_name=self.TEST_BUCKET_NAME,
+            path_in_bucket=self.FILENAME,
+        )
+
+        self.assertTrue(len(metadata) > 0)
```

Comment:
I'd like to see a roundtrip metadata test; cluster, sequence, tags uploaded and attached to the object in the store.... then download and instantiate a new datafile from the store object and check its tags etc are legit.